### PR TITLE
implement ROS system list using InventoryTable 

### PR DESCRIPTION
**Pull request by kgaikwad**
+ adding @karelhala into loop for review.

Pending work:
- [x] Expandable view
- [ ] _Not able to set default per_page=10, currently it takes `50` from inventory component (Not going to implement, keeping 50 as default per_page value)_
- [x] Sorting
- [x] Remove all dead code from existing table & pagination
- [x] set default ordering by `display_name` for inventory table 
- [x] Implement basic filtering by `display_name` column
- [x] add getUser() before ROS systems API
- [x] send multiple `getEntities` call when per_page > 50 as I can't send more than 50 UUIDs
- [x] add new route `/:inventoryId` which redirect to `RosDetailPage` and fix redirection link in `Name` column of list

### CODE REVIEW

**Comment by kgaikwad** [comment]
@karelhala,  I was able to render an expanded view per row. Though some refactoring required in the current code, pushing the expanded view changes.

**Comment by kgaikwad** [comment]
@karelhala, thank you for pointers. I am able to set default ordering by `display_name ASC`  instead of `updated` column.
Updated PR with this sorting change.

File: src/Routes/RosPage/RosPage.js [hunk 1]
```diff
@@ -26,25 +52,50 @@ class RosPage extends React.Component {
...
+                { key: 'state', title: 'State', props: { isStatic: true } }
+            ]
         };
 
+        this.inventory = React.createRef();
+        this.fetchSystems = this.fetchSystems.bind(this);
     }
 
-    async componentDidMount() {
-        await window.insights.chrome.auth.getUser();
```
Comment (kgaikwad) [inline src/Routes/RosPage/RosPage.js:36]: @karelhala, I have removed `getUser()` call. Please confirm if it is expected after adding changes with `InventoryTable` component.

Comment (karelhala) [reply src/Routes/RosPage/RosPage.js:36]: You don't need to do that on render, however you should call this before calling the `fetch` for ROS systems.

**Comment by kgaikwad** [comment]
Hello @terezanovotna,
Sharing a screenshot of latest ROS systems list using a `InventoryTable` component change. It includes list with pagination, sorting, basic filtering using display_name. Please feel free to share your suggestions/comments.

Screenshot from 2021-03-24 17-52-20

**Comment by kgaikwad** [comment]
Hello @karelhala, 
Updated PR with changes where it gives call to `getUser()` before ROS systems API and sends multiple getEntities calls when per_page > 50 as I can't send more than 50 UUIDs.

**Comment by karelhala** [comment]
@kgaikwad I don't think there's a real reference to this problem, it's just limitation of URL length supported by the server which is the limitation to this. The 50 UUIDs is just bellow the maximum number with all filters and pagination enabled.

File: src/Routes/RosPage/RosPage.js [hunk 2]
```diff
@@ -26,25 +51,92 @@ class RosPage extends React.Component {
...
-        this.props.fetchSystems({ page: this.state.page, perPage: this.state.perPage });
+        let params = {};
+        params.limit = fetchParams.perPage;
+        params.offset = (fetchParams.page - 1) * fetchParams.perPage;
+        params.order_by = fetchParams.orderBy || this.state.orderBy; /* eslint-disable-line camelcase */
+        params.order_how = fetchParams.orderHow || this.state.orderDirection; /* eslint-disable-line camelcase */
+
+        if (fetchParams.filters && fetchParams.filters.hostnameOrId) {
+            params.display_name =  fetchParams.filters.hostnameOrId; /* eslint-disable-line camelcase */
+        }
```
Comment (karelhala) [inline src/Routes/RosPage/RosPage.js:83]: This is rather expensive as it requires multiple assignments and memory calculation, much cleaner and faster approach is by assigning directly to params
```suggestion
        const params = {
          limit: fetchParams.perPage,
          offSet: (fetchParams.page - 1) * fetchParams.perPage,
          order_by: fetchParams.orderBy || this.state.orderBy, /* eslint-disable-line camelcase */
          order_how: fetchParams.orderHow || this.state.orderDirection, /* eslint-disable-line camelcase */
          ...fetchParams?.filters?.hostnameOrId && {
            display_name: fetchParams.filters.hostnameOrId; /* eslint-disable-line camelcase */
          }
        };
```

**Comment by kgaikwad** [comment]
@karelhala, removed WIP tag from PR. 
I have covered all points mentioned in PR description. Please request to review.

**Review by karelhala — APPROVED** [review]

**Comment by kgaikwad** [comment]
Thank you @karelhala for review. 
Merging these changes.

**Pull request merged**